### PR TITLE
feat(Product CRUD): CRUD operation of product

### PR DESCRIPTION
**Pull request by TuyishimireEric**
#### What does this PR do?

CRUD operation of a product
#### Description of Task to be completed?

the user should be able to add/create items for sale
Such that the stock can be seen by any requesting client

#### How should this be manually tested?

- run `npm install` 
- run `npm run migrate`
- run `npm run seed`
- run `npm run dev`
- check the [swagger documentation ](http://localhost:3000/swagger/)

#### Any background context you want to provide?

-dummy data are provided before having the real data

#### What are the relevant pivotal tracker/Trello stories?

- id of the project #187300155

### CODE REVIEW

File: src/controllers/productController.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        productExist,
+        'Product already exists, update instead!'
+      );
+    }
+
+    const images = await uploadImages();
+
+    const product = Database.Product.build({
+      name,
+      slug: name.split(' ').join('-'),
```
Comment (fab-ryan) [inline src/controllers/productController.ts:51]: create separate function it's will better in looking and check if the name has special characteres  ignore them

File: src/database/models/index.ts [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+
+  Product.belongsTo(User, {
+    foreignKey: 'sellerId',
+    as: 'seller',
+  });
+
+  User.hasMany(Product, {
+    foreignKey: 'sellerId',
+    as: 'products',
+  });
```
Comment (fab-ryan) [inline src/database/models/index.ts]: add this in each model it's will be readable

File: src/helper/buildQuery.ts
Comment (fab-ryan) [inline src/helper/buildQuery.ts:1]: it's seems this file is for only product while we can use it in different getting or update request try to make it resuable by any modal

File: src/helper/excOperation.ts
Comment (fab-ryan) [inline src/helper/excOperation.ts:1]: this is one is good

**Review by fab-ryan — CHANGES_REQUESTED** [review]
also add pagination helper function

File: src/database/models/index.ts (hunk 2 above)
Comment (fab-ryan) [reply src/database/models/index.ts]: please work on this

File: src/helper/excOperation.ts
Comment (TuyishimireEric) [reply src/helper/excOperation.ts:1]: Thank you

**Review by Mukunzijames — APPROVED** [review]

**Review by Ruthuwamahoro — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]
good work

**Pull request merged**